LEGO1/lego/legoomni/src/race/carrace.cpp

```diff
@@ -1,60 +1,418 @@
+	CarraceScript::c_srt001sl_RunAnim,
+	CarraceScript::c_srt002sl_RunAnim,
+	CarraceScript::c_srt003sl_RunAnim,
+	CarraceScript::c_srt004sl_RunAnim,
+	CarraceScript::c_srt005sl_RunAnim,
+	CarraceScript::c_srt001rh_RunAnim,
+	CarraceScript::c_srt002rh_RunAnim,
+	CarraceScript::c_srt003rh_RunAnim
+};
+
+// GLOBAL: LEGO1 0x100d5d30
+const MxS32 CarRace::g_unk0x100d5d30[] = {
+	CarraceScript::c_srt011sl_RunAnim,
+	CarraceScript::c_srt012sl_RunAnim,
+	CarraceScript::c_srt013sl_RunAnim,
+	CarraceScript::c_srt014sl_RunAnim
+};
+
+// GLOBAL: LEGO1 0x100d5d40
+const MxS32 CarRace::g_unk0x100d5d40[] =
+	{CarraceScript::c_srt015sl_RunAnim, CarraceScript::c_srt016sl_RunAnim, CarraceScript::c_srt017sl_RunAnim};
+
+// GLOBAL: LEGO1 0x100d5d50
+const MxS32 CarRace::g_unk0x100d5d50[] =
+	{CarraceScript::c_srt007rh_RunAnim, CarraceScript::c_srt008rh_RunAnim, CarraceScript::c_srt009rh_RunAnim};
+
+// GLOBAL: LEGO1 0x100d5d60
+const MxS32 CarRace::g_unk0x100d5d60[] =
+	{CarraceScript::c_srt010rh_RunAnim, CarraceScript::c_srt011rh_RunAnim, CarraceScript::c_srt012rh_RunAnim};
+
+// GLOBAL: LEGO1 0x100f0c70
+// STRING: LEGO1 0x100f0c48
+const LegoChar* g_strCRCFRNTY6 = "C_RCFRNTY6";
+
+// GLOBAL: LEGO1 0x100f0c74
+// STRING: LEGO1 0x100f0c3c
+const LegoChar* g_strCRCEDGEY0 = "C_RCEDGEY0";
+
+// GLOBAL: LEGO1 0x100f0c7c
+static MxS32 g_unk0x100f0c7c = 2;
+
 // FUNCTION: LEGO1 0x10016a90
 CarRace::CarRace()
 {
 	this->m_skeleton = NULL;
 	this->m_unk0x130 = MxRect32(0x16c, 0x154, 0x1ec, 0x15e);
 }
 
-// STUB: LEGO1 0x10016ce0
+// FUNCTION: LEGO1 0x10016ce0
+// FUNCTION: BETA10 0x100c8364
 MxResult CarRace::Create(MxDSAction& p_dsAction)
 {
+	MxResult result = LegoRace::Create(p_dsAction);
+
+	NavController()->SetDeadZone(1);
+	NavController()->SetTrackDefault(0);
+	GameState()->m_currentArea = LegoGameState::e_carrace;
+	GameState()->StopArea(LegoGameState::e_undefined);
+
+	LegoGameState* state = GameState();
+
+	RaceState* raceState = (RaceState*) state->GetState("CarRaceState");
+
+	if (!raceState) {
+		raceState = (RaceState*) state->CreateState("CarRaceState");
+	}
+
+	m_raceState = raceState;
+
+	m_act1State->m_unk0x018 = 6;
+	m_unk0x144 = -1;
+	m_unk0x148 = -1;
+	m_unk0x14c = -1;
+
+	LegoRaceCar::FUN_10012e00();
+
+	MxS32 streamId =
+		DuneBuggy::GetColorOffset(g_strCRCEDGEY0) + (DuneBuggy::GetColorOffset(g_strCRCFRNTY6) * 5 + 15) * 2;
+	InvokeAction(Extra::e_start, m_atomId, streamId, NULL);
+	InvokeAction(Extra::e_start, m_atomId, CarraceScript::c_RaceCarDashboard, NULL);
+
 	// TODO
 	return SUCCESS;
 }
 
-// STUB: LEGO1 0x10016dd0
+// FUNCTION: LEGO1 0x10016dd0
+// FUNCTION: BETA10 0x100c8490
 void CarRace::ReadyWorld()
 {
-	// TODO
+	assert(m_hideAnim);
+	LegoWorld::ReadyWorld();
+	m_hideAnim->FUN_1006db40(0);
+
+	MxDSAction action;
+	action.SetAtomId(*g_jukeboxScript);
+	action.SetObjectId(16);
+
+	BackgroundAudioManager()->PlayMusic(action, 5, MxPresenter::e_repeating);
+	AnimationManager()->Resume();
+	FUN_10015820(FALSE, 7);
```

Comment:
The constant `7` should be replaced with `LegoOmni::c_disableInput | LegoOmni::c_disable3d | LegoOmni::c_clearScreen`. There's also another in `legocarbuild.cpp` where you could fix that. Thank you!